Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -304,23 +305,20 @@ WebFramePolicyListenerProxy& WebFrameProxy::setUpPolicyListenerProxy(CompletionH
 
 void WebFrameProxy::getWebArchive(CompletionHandler<void(API::Data*)>&& callback)
 {
-    if (!m_page)
-        return callback(nullptr);
-    m_page->getWebArchiveOfFrame(this, WTFMove(callback));
+    if (RefPtr page = m_page.get())
+        page->getWebArchiveOfFrame(this, WTFMove(callback));
 }
 
 void WebFrameProxy::getMainResourceData(CompletionHandler<void(API::Data*)>&& callback)
 {
-    if (!m_page)
-        return callback(nullptr);
-    m_page->getMainResourceDataOfFrame(this, WTFMove(callback));
+    if (RefPtr page = m_page.get())
+        page->getMainResourceDataOfFrame(this, WTFMove(callback));
 }
 
 void WebFrameProxy::getResourceData(API::URL* resourceURL, CompletionHandler<void(API::Data*)>&& callback)
 {
-    if (!m_page)
-        return callback(nullptr);
-    m_page->getResourceDataFromFrame(*this, resourceURL, WTFMove(callback));
+    if (RefPtr page = m_page.get())
+        page->getResourceDataFromFrame(*this, resourceURL, WTFMove(callback));
```

Comment:
The old code called `callback(nullptr)` when `m_page` is null. Did you remove that intentionally?